test/test-helpers.h

```diff
@@ -207,6 +207,95 @@ class ExceptionWithMsg : public Catch::MatcherBase<Exception> {
 	}
 };
 
+/* \brief Automatically restores environment variable to its original state
+ * when the test finishes running.
+ *
+ * When an EnvVar object is created, it remembers the state of the given
+ * environment variable. After that, the test can safely change the variable
+ * directly through setenv(3) and unsetenv(3), or via convenience methods set()
+ * and unset(). When the test finishes (with whatever result), the EnvVar
+ * restores the variable to its original state.
+ *
+ * If you need to run some code after the variable is changed (e.g. tzset(3)
+ * after a change to TZ), use `on_change()` method to specify a function that
+ * should be ran. Note that it will be executed only if the variable is changed
+ * through set() and unset() methods; EnvVar won't notice a change made using
+ * setenv(3), unsetenv(3), or a change made by someone else (e.g. different
+ * thread).
+ */
+class EnvVar {
+	std::function<void(void)> on_change_fn;
+	std::string name;
+	std::string value;
+	bool was_set = false;
+
+public:
+	/// \brief Safekeeps the value of environment variable \a name.
+	///
+	/// \note Accepts a string by value since you'll probably pass it
+	/// a temporary or a string literal anyway. Just do it, and use set() and
+	/// unset() methods, instead of keeping a local variable with a name in it
+	/// and calling setenv(3) and unsetenv(3).
+	EnvVar(std::string name_)
+		: name(std::move(name_))
+	{
+		const char* original = ::getenv(name.c_str());
+		was_set = original != nullptr;
+		if (was_set) {
+			value = std::string(original);
+		}
+	}
+
+	~EnvVar()
+	{
+		if (was_set) {
+			set(value);
+		} else {
+			unset();
+		}
+	}
+
+	/// \brief Changes the value of the environment variable.
+	///
+	/// \note This is equivalent to you calling setenv(3) yourself.
+	///
+	/// \note This does \emph{not} change the value to which EnvVar will
+	/// restore the variable when the test finished running. The variable is
+	/// always restored to the state it was in when EnvVar object was
+	/// constructed.
+	void set(std::string new_value) const
+	{
+		const auto overwrite = true;
+		::setenv(name.c_str(), new_value.c_str(), (int)overwrite);
```

Comment:
Yeah, I'm picky, but... static_cast instead of C cast? or drop casting all together.

Reply:
I actually forgot that `bool` is implicitly convertible to `int` ._. I'll remove the cast throughout. Thank you!